packages/base-dao/src/baseDAO.ts

```diff
@@ -367,20 +367,23 @@ export function createEntityDAO({ entityName, hooks, pluralizationFunction = plu
       };
       return dataProvider.getEntitiesConnection(entityName, transformedArgs);
     },
-    getRelatedEntityId: async (originalEntityId: string, relationEntityName: string) => {
+    getRelatedEntityId: async (originalEntityId: string, relationEntityName: string): Promise<string | null> => {
       const relation = await dataLoaderForSingleRelatedEntity.load({ originalEntityId, relationEntityName });
       return relation && relation.id;
     },
-    getRelatedEntity: async (originalEntityId: string, relationEntityName: string) => {
+    getRelatedEntity: async <T>(originalEntityId: string, relationEntityName: string): Promise<T | null> => {
       return dataLoaderForSingleRelatedEntity.load({ originalEntityId, relationEntityName });
```

Comment:
Since the base dao type is any, it won't allow passing the generics to the underlying base dao